chain/chain/src/flat_storage_resharder.rs

```diff
@@ -222,6 +222,13 @@ impl FlatStorageResharder {
                 split_params.clone(),
             )),
         );
+        // Do not update parent flat head, to avoid overriding the resharding status.
+        // In any case, at the end of resharding the parent shard will completely disappear.
+        self.runtime
```

Comment:
I had to freeze the parent flat storage in the end, because otherwise something in chain was overriding its status.